lib/avo/fields/has_base_field.rb

```diff
@@ -91,6 +96,16 @@ def authorized?
           true
         end
       end
+
+      def name
+        super(from_use_resource: use_resource&.name)
+      end
```

Comment:
I was thinking of a different route where we didn't have to introduce `from_use_resource` to the `BaseField.name` method.

```suggestion
      def name
        return use_resource&.name if use_resource.present?
      
        super
      end
```

Reply:
That approach have an issue, if use resource is present the `name: "CustomName"` and translation will be unconsidered. I don't think we want it, we want the name to have the flowing priority:

1. Custom name, using name option
2. Translation keys
3. Name from use resource if present, otherwise
4. Name from id

I understand that my first approach and second one are kinda tricky but that was the ways i found to maintain this naming priority to be the same without duplicating the code and the guards

Reply:
Check the last approach based on what you suggested